File: elixir/log-level/lib/log_level.ex
defmodule LogLevel do
  def to_label(level, legacy?) do
    cond do
      # note to self: use === (or == if floats are ok) here, not =
      level === 0 and not legacy? -> :trace
      level === 1 -> :debug
      level === 2 -> :info
      level === 3 -> :warning
      level === 4 -> :error
      level === 5 and not legacy? -> :fatal
      true -> :unknown
    end
  end

  def alert_recipient(level, legacy?) do
    label = to_label(level, legacy?)
    cond do
      label === :error or label === :fatal -> :ops
      label === :unknown and legacy? -> :dev1
      true -> :dev2

   #   label === :unknown and not legacy? -> :dev2
   #   true -> false
    end
  end
end

File: java/logs-logs-logs/src/main/java/LogLevel.java
public enum LogLevel {
    TRACE(  "TRC", 1),
    DEBUG(  "DBG", 2),
    INFO(   "INF", 4),
    WARNING("WRN", 5),
    ERROR(  "ERR", 6),
    FATAL(  "FTL", 42),
    UNKNOWN("UNK", 0);

    private final String abbr;
    private final int num;

    LogLevel(String abbreviation, int numericLevel) {
        abbr = abbreviation;
        num = numericLevel;
    }

    public String getAbbreviation() {
        return abbr;
    }

    public int getNumericLevel() {
        return num;
    }

    public static LogLevel getLevel(String abbreviation) {
        for (LogLevel level : values())
            if (level.getAbbreviation().equals(abbreviation))
                return level;
        return UNKNOWN;
    }
}
